### ISSUE TITLE
Problem with lofar libraries

### ISSUE THREAD

**Issue by nudomarinero** [kernsuite/packaging#96]
In a clean Ubuntu 16.04 installation I got the following error after installing "lofar" (from Kern-3):
```
$ msoverview in=L136064_SAP000_SB041_uv.MS
msoverview: Version 20110407GvD
Error: Shared library lofarstman not found in CASACORE_LDPATH or (DY)LD_LIBRARY_PATH
libcasa_lofarstman.so.2: cannot open shared object file: No such file or directory
libcasa_lofarstman.so: cannot open shared object file: No such file or directory
liblofarstman.so.2: cannot open shared object file: No such file or directory
liblofarstman.so: cannot open shared object file: No such file or directory
```
The problem is solved installing the package "lofar-dev".

**Comment by gijzelaerr** [comment]
This is a problem with LOFAR software itself, it expects the development links for libraries installed. I don't want to add the `lofar-dev` dependency to it by default since the package itself is already quite big. Maybe it helps if you report it upstream (again).

**Comment by gijzelaerr** [comment]
wait let me leave this issue open and add a label, that is more useful.

**Comment by tammojan** [comment]
Raw LOFAR measurement sets do not work without the dynamically loaded shared library being installed. This is by design. If this shared library is part of `lofar-dev`, then apparently you need to install that.

**Comment by gijzelaerr** [comment]
they are installed, but only the real library, not all the symlinks. The symlinks are supposed to be used during development. An example:
```
liblofarstman0 contains /usr/lib/liblofarstman.so.0
lofar-dev contains /usr/lib/liblofarstman.so which is a symlink to the .so.0 file
```
this is the same for *all* debian packages.

The lofar package depends on for `liblofarstman0` and others, so the shared library is available, `msoverview` is looking for the wrong path.

**Comment by gijzelaerr** [comment]
When I look at it again now I see that `msoverview` tries to load SO version specific libraries first, but the version mismatch.

I think this has to do with this applied patch that is needed that never reached upstream:

https://github.com/kernsuite-debian/lofar/blob/master/debian/patches/soversion

I set the version to 0, since at the moment no versioning seems to be used upstream at all. So I'm surprised `msoverview` tries to use version 2? I guess it is time to start to do proper SO versioning upstream.

**Comment by gijzelaerr** [comment]
@tammojan what about this one? It looks like `msoverview` expects an SO version set to 2? What is the reasoning?

**Comment by gijzelaerr** [comment]
@tammojan any update on this?

**Comment by tammojan** [comment]
The reason msoverview looks for version 2 is that casacore is at major version 2. The dynamic library is not searched for by msoverview itself, but from the underlying casacore tables layer. The relevant line in casacore is [here](https://github.com/casacore/casacore/blob/master/tables/DataMan/DataManager.cc#L326). Casacore's `SOVERSION` is by default the project major version, see [here](https://github.com/casacore/casacore/blob/master/CMakeLists.txt#L480). So 2 is a better number than 0 here.

**Comment by gijzelaerr** [comment]
ok, then I'll set this to 2. Is this never going to reach upstream?

**Comment by gijzelaerr** [comment]
meanwhile, this has been fixed in 3.1.5-1 which has been uploaded to KERN-dev and will land in KERN-5. @tammojan let me know about merging in upstream (should still happen) but for now I close this issue.

**Comment by tammojan** [comment]
I reported it even upstreamer, so it may well be merged. I'll report it here when it's merged.
